app/controllers/concerns/hashed_emails.rb

```diff
@@ -0,0 +1,13 @@
+module HashedEmails
+  GIT_COOKIE_PREFERENCES = "git-cookie-preferences-v1".freeze
+
+  def hash_email_address?
+    return unless cookies[GIT_COOKIE_PREFERENCES]
+
+    JSON.parse(cookies[GIT_COOKIE_PREFERENCES]).then do |prefs|
+      prefs["functional"] && prefs["marketing"]
```

Comment:
I think this is why it is not working quite right @martyn-w - I think you are confusing `functional` and `non-functional`?

* `functional` cookies are the essential ones
* `non-functional` cookies are the analytics ones
* `marketing` cookies are the marketing ones (obviously)

For example, if I reject all cookies, `functional` is still set to `true`:
![image](https://github.com/user-attachments/assets/486265f4-8497-4440-9e6b-c366260a914a)


We should only be hashing if `non-functional` and `marketing` are both `true`

Reply:
Hi Gemma, you are absolutely correct - apologies for this, I'm fixing it now